Update multi-label instance creation

mberr opened the pull request:
Create LCWA triples as [CSR matrix](http://scipy-lectures.org/advanced/scipy_sparse/csr_matrix.html) of shape `(num_pairs, num_entities)` using only numpy/scipy routines without any Python loops. 

Also remove unused attributes from training `Instances`, and cleanup class structure.

mberr committed:
Add torch-based method to create LCWA instances

mberr committed:
Fix TriplesNumericLiteralsFactory

mberr committed:
Some fixes

mberr committed:
Remove unused relation_to_id mapping from Instances

mberr committed:
Simplify instances

mberr committed:
Remove unused method

mberr committed:
Fix LCWA triples generation

mberr committed:
Extend tests

cthoyt commented on src/pykeen/triples/instances.py:
```diff
@@ -27,34 +27,22 @@
+
+    def __len__(self):  # noqa: D105
+        return self.mapped_triples.shape[0]
+
```
this one should also get a `from_triples` class method even though it will be trivial

cthoyt replied:
okay maybe this isn't important since there's an adapter pattern used to instantiate these

mberr replied:
Also, we already have a kind of factory, when creating them with `create_*_instance` from the TriplesFactory and/or TrainingLoop

cthoyt commented on src/pykeen/triples/instances.py:
```diff
@@ -64,20 +52,50 @@ def __getitem__(self, item):  # noqa: D105
 class MultimodalInstances(Instances):
     """Triples and mappings to their indices as well as multimodal data."""
 
+    #: TODO: do we need these?
```
it remains a great mystery since we don't have any unit tests running for the literal models

mberr replied:
Yes, being required to apply any changes also to the literal models without having any way to validate them is also a problem for me :wink: I already thought about suggesting to remove them until they come back with a justification (=test / training script / ...).

cthoyt replied:
okay well good news is the tests for multimodal are still passing, so you can have some confidence to start wrecking things ;)

PyKEEN-bot committed:
Trigger CI

mberr committed:
Merge remote-tracking branch 'origin/master' into update-multi-label-instance-creation

cthoyt committed:
Pass flake8

Trigger CI

cthoyt committed:
Update type hints

cthoyt committed:
Update old attributes

@mberr is this correct? This attribute isn't there anymore, and it looks to me like it's the same as the __len__

PyKEEN-bot committed:
Trigger CI

mberr committed:
Adjust test for changed structure

mberr committed:
Merge remote-tracking branch 'origin/master' into update-multi-label-instance-creation

PyKEEN-bot committed:
Trigger CI
